File: com-mini-mvc-test/src/main/java/com/mini/web/test/dao/impl/UserDaoImpl.java
package com.mini.web.test.dao.impl;

import com.mini.jdbc.JdbcTemplate;
import com.mini.web.test.dao.UserDao;

import javax.inject.Inject;
import javax.inject.Named;
import javax.inject.Singleton;

/**
 * UserDaoImpl.java
 * @author xchao
 */
@Singleton
@Named("userDao")
public class UserDaoImpl implements UserDao {
    private JdbcTemplate jdbcTemplate;

    @Inject
    public void setJdbcTemplate(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @Override
    public JdbcTemplate writeTemplate() {
        return jdbcTemplate;
    }

    @Override
    public JdbcTemplate readTemplate() {
        return jdbcTemplate;
    }
}

File: com-mini-mvc-kotlin/src/main/kotlin/com/mini/mvc/kotlin/dao/impl/UserDaoImpl.kt
package com.mini.mvc.kotlin.dao.impl

import com.mini.jdbc.JdbcTemplate
import com.mini.mvc.kotlin.dao.UserDao
import com.mini.mvc.kotlin.entity.mapper.UserMapper
import javax.inject.Inject
import javax.inject.Named
import javax.inject.Singleton

/**
 * UserDaoImpl.java
 * @author xchao
 */
@Singleton
@Named("userDao")
@Suppress("unused")
class UserDaoImpl : UserDao {

    @set:Inject
    var jdbcTemplate: JdbcTemplate? = null

    @set:Inject
    override var userMapper: UserMapper? = null


    override fun writeTemplate(): JdbcTemplate? {
        return jdbcTemplate
    }

    override fun readTemplate(): JdbcTemplate? {
        return jdbcTemplate
    }
}
